src/coreclr/jit/flowgraph.cpp

```diff
@@ -916,8 +916,7 @@ GenTreeCall* Compiler::fgGetSharedCCtor(CORINFO_CLASS_HANDLE cls)
         CORINFO_RESOLVED_TOKEN resolvedToken;
         memset(&resolvedToken, 0, sizeof(resolvedToken));
         resolvedToken.hClass = cls;
-
-        return impReadyToRunHelperToTree(&resolvedToken, CORINFO_HELP_READYTORUN_STATIC_BASE, TYP_BYREF);
+        return impReadyToRunHelperToTree(&resolvedToken, m_preferredInitCctor, TYP_BYREF);
```

Comment:
Can you please ensure we create this as `TYP_VOID` when `m_preferredInitCctor == CORINFO_HELP_READYTORUN_CCTOR_TRIGGER`?